torch/csrc/jit/script/sugared_value.h

```diff
@@ -177,6 +191,53 @@ struct TORCH_API BuiltinFunction : public SugaredValue {
       c10::optional<NamedValue> self);
 };
 
+struct TORCH_API SugaredTupleValue : public SugaredValue {
+  explicit SugaredTupleValue(
+      std::vector<std::shared_ptr<SugaredValue>> tup,
+      bool contains_module_list)
+      : tup_(tup), contains_module_list_(contains_module_list){};
+
+  std::vector<std::shared_ptr<SugaredValue>> asTuple(
+      const SourceRange& loc,
+      Function& m,
+      const c10::optional<size_t>& size_hint = {}) override {
+    return tup_;
+  };
+
+  Value* asValue(const SourceRange& loc, Function& m) override {
+    std::vector<Value*> vec;
+    for (const auto& sv : tup_) {
+      vec.push_back(sv->asValue(loc, m));
+    }
+    Graph& g = *m.graph();
+    return g.insertNode(g.createTuple(vec))->output();
+  }
+
+  std::string kind() const override {
+    return "Sugared Tuple";
+  }
+
+  SugaredValuePtr getitem(const SourceRange& loc, Function& m, Value* idx)
+      override {
+    TORCH_INTERNAL_ASSERT(
+        toIValue(idx), loc, "Expected integer literal for Sugared Tuple");
+    auto index = toIValue(idx)->toInt();
+    TORCH_INTERNAL_ASSERT(
+        index >= 0 && index < static_cast<int64_t>(tup_.size()),
+        loc,
+        "Index out of range of Sugared Tuple");
+    return tup_.at(index);
+  }
+
+  IterableValuePtr asIterable(const SourceRange& loc, Function& m) override {
+    return std::make_shared<IterableValue>(
+        shared_from_this(), tup_.size(), contains_module_list_);
+  };
+
+  std::vector<std::shared_ptr<SugaredValue>> tup_;
+  bool contains_module_list_;
```

Comment:
This can use more explanation. What are the semantics for when we unroll or do not unroll a loop? Can that be put in a comment somewhere.